temporal-serviceclient/src/main/java/io/temporal/internal/retryer/GrpcRetryerUtils.java

```diff
@@ -67,8 +66,14 @@ class GrpcRetryerUtils {
         // never retry these codes
         return currentException;
       case INTERNAL:
+        GetSystemInfoResponse.Capabilities capabilities;
+        try {
+          capabilities = serverCapabilities.get();
```

Comment:
Sorry I don't understand this change, why do we want to catch every exception from `serverCapabilities.get()`? and if we do why only in this one place?

Reply:
`serverCapabilities.get()` performs a memoized `WorkflowService.getSystemInfo` gRPC call.  That can throw on the first call, or if all previous calls also threw.  If it throws, it will not be caught and retried, but instead will fail the entire operation (for which `WorkflowService.getSystemInfo` is almost irrelevant except for one error-handling branch).

Reply:
Yeah I think failing the entire operation because `WorkflowService.getSystemInfo` failed is wrong, totally agree with you. What I am not sure is if this is the right way to solve the problem. We call `getSystemInfo` in other places, like starting a worker, do we need to do anything to those callers?

I can check how Go handle this, @Sushisource How does core handle `getSystemInfo` and specifically failures on `WorkflowService.getSystemInfo`? Does it retry it like any other gRPC request?

Reply:
Huh, interesting. I would've said it retries, but, after looking it up it doesn't. Trivial to change though (and probably should be). You can see here we make the call before putting the client inside the retry wrapper: https://github.com/temporalio/sdk-core/blob/bcda5d9f73b082ddf6816921e15db5e68a8ce47e/client/src/lib.rs#L430

Reply:
How would you handle it in core we should do the same thing in Java IMO. What I would expect is `get_system_info` is retried by the normal retry with a set deadline (5s?) and the retry has logic to avoid the recursion of `get_system_info` being needed by the retrier?

Reply:
Yes. If all I do is just move that call to happen from the `RetryClient`, that will all work (though the default deadline is 30s). I could change the deadline to be shorter specifically for that call.